lib/oppen.rb

```diff
@@ -22,4 +22,91 @@ def self.print(tokens: [], margin: 80, new_line: "\n")
     end
     printer.output
   end
+
+  # Wadler.
+  class Wadler
+    attr_reader :tokens
+    attr_reader :margin
+    attr_reader :new_line
+    attr_reader :current_nest_indentation
+
+    def initialize(margin: 80, new_line: "\n")
+      @tokens = []
+      @margin = margin
+      @new_line = new_line
+      @current_nest_indentation = 0
+    end
+
+    # <Description>
+    # @return [<Type>] <description>
+    def output
+      tokens.append(::Oppen::Token::EOF.new)
+      ::Oppen.print(tokens:, margin:, new_line:)
+    end
+
+    # <Description>
+    # @param indent [<Type>] <description>
+    # @param open_obj [<Type>] <description>
+    # @param close_obj [<Type>] <description>
+    # @param break_type [<Type>] <description>
+    #
+    # @return [<Type>] <description>
+    def group(indent: 0, open_obj: '', close_obj: '',
+              break_type: ::Oppen::Token::BreakType::CONSISTENT)
+      if open_obj != ''
+        self.break
+        text(open_obj)
+      end
+
+      nest(indent:, break_type:) { # rubocop:disable Style/ExplicitBlockArgument
+        yield
+      }
+
+      if close_obj != '' # rubocop:disable Style/GuardClause
+        self.break
+        text(close_obj)
+      end
+    end
+
+    # <Description>
+    # @param indent [<Type>] <description>
+    # @param break_type [<Type>] <description>
+    #
+    # @return [<Type>] <description>
+    def nest(indent: 2, break_type: ::Oppen::Token::BreakType::CONSISTENT)
+      last_nest_indentation = current_nest_indentation
+      @current_nest_indentation = indent
+
+      tokens.append(::Oppen::Token::Begin.new(offset: indent, break_type:))
+
+      yield
+
+      tokens.append(::Oppen::Token::End.new)
+      @current_nest_indentation = last_nest_indentation
+    end
```

Comment:
Now that we're [dogfooding](https://en.wikipedia.org/wiki/Eating_your_own_dog_food), I think it's clear that the API is not that nice.

What about we add some module methods `Oppen.begin(…)`, `Oppen.consistent(…)`, etc. ?